apps/aecore/src/aec_conductor.erl

```diff
@@ -1478,14 +1525,37 @@ handle_add_block(Header, Block, State, Origin) ->
             handle_add_block(Block, Hash, Prev, State, Origin)
     end.
 
-handle_add_block(Block, Hash, Prev, #state{top_block_hash = TopBlockHash} = State, Origin) ->
+%% Used in PoS hyperchains to allow invalid_leader error and not crash the conductor.
+%% This error is still fatal in other chain types.
+safe_insert_block(aec_consensus_hc, Block, Origin) ->
+    try
+        aec_chain_state:insert_block_conductor(Block, Origin)
+    catch exit:{aborted, {{leader_validation_failed, invalid_leader} = Error, _Stack1}}:_Stack2 ->
+        %% This error is not fatal in PoS hyperchains, the block which failed validation should be ignored
+        %% Convert exception to an error value
+        {error, Error}
+    end;
```

Comment:
Similar feeling about this one, why would we produce blocks with the wrong leader? And if we do, why should it fail in a different way for PoS?

Removing the `aec_consensus_hc` clause does not make the HC-suite fail 🤔